lib/ecto/changeset.ex

```diff
@@ -729,6 +738,9 @@ defmodule Ecto.Changeset do
         case Map.get(types, key) do
           {:embed, embed} ->
             {key, Ecto.Embedded.apply_changes(embed, value)}
+          {:assoc, _assoc} ->
+            raise ArgumentError, "associations should be removed from changeset" <>
```

Comment:
If so, this would definitely be possible to do, we just need to make sure we strip all associations from changes before we call `apply_changes/1` inside the repository.